spec/activerecord-bitemporal/date_type_valid_time_spec.rb

```diff
@@ -0,0 +1,1112 @@
+      it { is_expected.to be_falsey }
+    end
+  end
+
+  describe "Association" do
+    describe "Date type valid time has many Date type vaild time" do
+      describe "sync valid_from in create" do
+        let!(:cop1) { department.cops.new(name: "Jane") }
+        let!(:cop2) { department.cops.new(name: "Homu") }
+        let!(:cop3) { department.cops.new(name: "Mami", valid_from: "2016/01/15") }
+
+        context "#new to #save" do
+          let(:department) { Department.new(name: "Department") }
+
+          context "saving" do
+            subject { department.save }
+            it { expect { subject }.to change { cop1.valid_from } }
+            it { expect { subject }.to change { cop2.valid_from } }
+            it { expect { subject }.not_to change { cop3.valid_from } }
+          end
+
+          context "saved" do
+            subject { department.valid_from }
+            before { department.save }
+            it { is_expected.to eq cop1.valid_from }
+            it { is_expected.to eq cop2.valid_from }
+            it { expect(cop1.valid_from).to eq cop2.valid_from }
+          end
+
+          context "department with valid_from" do
+            let(:department) { Department.new(name: "Department", valid_from: "2018/12/25") }
+            subject { department.valid_from }
+            before { department.save }
+
+            it { is_expected.to eq "2018/12/25".to_date }
+            it { is_expected.to eq cop1.valid_from }
+            it { is_expected.to eq cop2.valid_from }
+            it { expect(cop1.valid_from).to eq cop2.valid_from }
+          end
+        end
+
+        context "#create to #save" do
+          let(:department) { Timecop.travel(1.day.ago) { Department.create(name: "Department") } }
+
+          context "saving" do
+            subject { department.save }
+            it { expect { subject }.to change { cop1.valid_from } }
+            it { expect { subject }.to change { cop2.valid_from } }
+            it { expect { subject }.not_to change { cop3.valid_from } }
+          end
+
+          context "saved" do
+            before { department.save }
+            subject { department.valid_from }
+            it { is_expected.not_to eq cop1.valid_from }
+            it { is_expected.not_to eq cop2.valid_from }
+            it { expect(cop1.valid_from).to eq cop2.valid_from } # same date
+          end
+        end
+      end
+
+      describe "inverse_of" do
+        let(:department) do
+          klass = Class.new(Department) {
+            has_many :cops, foreign_key: :department_id, inverse_of: :department
+
+            def self.name
+              "DepartmentWithInverseOf"
+            end
+          }
+          Timecop.travel(1.day.ago) { klass.create!(name: "Department") }
+        end
+        before do
+          department.update(name: "Department2")
+          department.cops.create!(name: "Jane")
+        end
+
+        it { expect(department.reload.cops.first.department).to be department }
```

Comment:
Since this is compared by the primary key, this test is not able to check that they are the same object. See https://github.com/rspec/rspec-rails/issues/1173

However, this is an issue in the original spec, so you can leave it as it is.

Reply:
This test uses `be` matcher.  `be` matcher is tests using the `equal?` method.  
https://github.com/rspec/rspec-expectations/blob/v3.12.3/lib/rspec/matchers.rb#L349-L351

`equal?` checks the equality of objects, so it checks that `department` is the same object here.
https://docs.ruby-lang.org/en/3.1/BasicObject.html#method-i-equal-3F